dlt/destinations/synapse/synapse.py

```diff
@@ -0,0 +1,142 @@
+from typing import ClassVar, Dict, Optional, Sequence, List, Any, Tuple
+
+from dlt.common.wei import EVM_DECIMAL_PRECISION
+from dlt.common.destination.reference import NewLoadJob
+from dlt.common.destination import DestinationCapabilitiesContext
+from dlt.common.data_types import TDataType
+from dlt.common.schema import TColumnSchema, TColumnHint, Schema
+from dlt.common.schema.typing import TTableSchema
+from dlt.common.utils import uniq_id
+
+from dlt.destinations.sql_jobs import SqlStagingCopyJob, SqlMergeJob
+
+from dlt.destinations.insert_job_client import InsertValuesJobClient
+
+from dlt.destinations.synapse import capabilities
+from dlt.destinations.synapse.sql_client import PyOdbcSynapseClient
+from dlt.destinations.synapse.configuration import SynapseClientConfiguration
+from dlt.destinations.sql_client import SqlClientBase
+
+
+SCT_TO_PGT: Dict[TDataType, str] = {
+    "complex": "nvarchar(4000)",
+    "text": "nvarchar(4000)",
+    "ntext": "nvarchar(4000)",
+    "double": "float",
+    "bool": "bit",
+    "timestamp": "datetimeoffset",
+    "date": "date",
+    "bigint": "bigint",
+    "binary": "varbinary(8000)",
+    "decimal": "decimal(%i,%i)",
+    "time": "time"
+}
+
+PGT_TO_SCT: Dict[str, TDataType] = {
+    "nvarchar": "text",
+    "float": "double",
+    "bit": "bool",
+    "datetimeoffset": "timestamp",
+    "date": "date",
+    "bigint": "bigint",
+    "varbinary": "binary",
+    "decimal": "decimal",
+    "time": "time"
+}
+
+HINT_TO_SYNAPSE_ATTR: Dict[TColumnHint, str] = {
+    "unique": "UNIQUE"
+}
+
+class SynapseStagingCopyJob(SqlStagingCopyJob):
+
+    @classmethod
+    def generate_sql(cls, table_chain: Sequence[TTableSchema], sql_client: SqlClientBase[Any]) -> List[str]:
+        sql: List[str] = []
+        for table in table_chain:
+            with sql_client.with_staging_dataset(staging=True):
+                staging_table_name = sql_client.make_qualified_table_name(table["name"])
+            table_name = sql_client.make_qualified_table_name(table["name"])
+            # drop destination table
+            sql.append(f"DROP TABLE {table_name};")
+            # moving staging table to destination schema
+            sql.append(f"ALTER SCHEMA {sql_client.fully_qualified_dataset_name()} TRANSFER {staging_table_name};")
+            # recreate staging table
+            sql.append(f"SELECT * INTO {staging_table_name} FROM {table_name} WHERE 1 = 0;")
```

Comment:
does this really only create the staging table schema, or does it also copy data? It should only do the former and not copy any data over.